builtin/logical/pkiext/pkiext_binary/acme_test.go

```diff
@@ -651,6 +652,43 @@ func SubtestACMEPreventsICADNS(t *testing.T, cluster *VaultPkiCluster) {
 	pki.RemoveDNSRecordsForDomain(hostname)
 }
 
+func SubtestACMEACMEshTLSALPN(t *testing.T, cluster *VaultPkiCluster) {
+	// Setup the mount
+	pki, err := cluster.CreateAcmeMount("pki-acmesh-tls-alpn")
+	require.NoError(t, err, "failed setting up acme mount")
+	directory := "https://" + pki.GetActiveContainerIP() + ":8200/v1/pki/acme/directory"
+
+	// Start the container
+	vaultNetwork := cluster.GetContainerNetworkName()
+	t.Logf("Starting acme.sh on network=%v", vaultNetwork)
+
+	runner, result := GetACMEshContainer(t, vaultNetwork)
+	require.NotNil(t, result, "failed to start container")
+	require.NotNil(t, runner, "failed to start container")
+	defer runner.Stop(context.Background(), result.Container.ID)
+
+	// Get and provision its IP address.
+	networks, err := runner.GetNetworkAndAddresses(result.Container.ID)
+	require.NoError(t, err, "failed to get container network IP address")
+
+	ipAddr := networks[vaultNetwork]
+	hostname := "acmesh-tls-alpn-check-acme-client.dadgarcorp.com"
+
+	err = pki.AddHostname(hostname, ipAddr)
+	require.NoError(t, err, "failed to update vault host files")
+
+	// Now try an issuance.
+	issueAlpnCmd := []string{
+		"bash", "-i", "-c", "./acme.sh", "--issue", "--alpn", "-d", hostname,
+		"--insecure", "--server", directory, "--debug",
+	}
+	ctx := context.Background()
```

Comment:
maybe we could reduce this to make testing a bit faster?  like WithDeadline?

Reply:
This does not fail, so this context shouldn't fail with a deadline. We rely on Go's testing default deadline here to catch this if it fails in a way that hangs. I don't think modifying this context really makes sense on our own, we're probably better off trying to see if `testing.T` can get context support based on global test timeouts.